src/Build.UnitTests/BackEnd/TargetBuilder_Tests.cs

```diff
@@ -866,6 +872,53 @@ public void TestAfterTargetsWithTwoReferringToOne()
 
             BuildResult result = builder.BuildTargets(GetProjectLoggingContext(entry), entry, this, entry.Request.Targets.ToArray(), CreateStandardLookup(project), CancellationToken.None).Result;
             AssertTaskExecutionOrder(new string[] { "BuildTask", "AfterTask", "AfterTask2" });
+            Assert.False(result.ResultsByTarget["Build"].AfterTargetsHaveFailed);
+        }
+
+        /// <summary>
+        /// Test a failing after target
+        /// </summary>
+        [Fact]
+        public void TestAfterTargetsWithFailure()
+        {
+            string projectBody = @"
+<Target Name='Build'>
+    <BuildTask/>
+</Target>
+
+<Target Name='After' AfterTargets='Build'>
+    <ErrorTask1/>
+</Target>
+";
+
+            BuildResult result = BuildSimpleProject(projectBody, new string[] { "Build" }, failTaskNumber: 2 /* Fail on After */);
+            Assert.Equal(TargetResultCode.Success, result.ResultsByTarget["Build"].ResultCode);
+            Assert.True(result.ResultsByTarget["Build"].AfterTargetsHaveFailed);
+        }
+
+        /// <summary>
+        /// Test a transitively failing after target
+        /// </summary>
+        [Fact]
+        public void TestAfterTargetsWithTransitiveFailure()
```

Comment:
Could you also add an equivalent test but which runs the build in an out of proc node? You can use [BuildManagerSession](https://github.com/microsoft/msbuild/blob/a1a53cb88da5d05eb854f02be5ee19edd0c2f399/src/Shared/UnitTests/ObjectModelHelpers.cs#L1863) and send it `BuildParameters.DisableInprocNode` to force builds to be out of proc.

Reply:
Turns out such tests already exist in BuildManager_Tests. I have tweaked them to also run with `BuildParameters.DisableInprocNode`. And verified that they were failing without the changes in this PR and with the flag set.